src/workerd/api/streams/standard.c++

```diff
@@ -1094,20 +1094,58 @@ namespace {
 
 class AllReaderBase {
 public:
+  class WeakRef : public kj::Refcounted {
```

Comment:
Could we use the already-existing `WeakRef<AllreaderBase>`? https://github.com/cloudflare/workerd/blob/main/src/workerd/api/streams/standard.h#L664-L676 `tryGetAs()` could just be a free function.

Reply:
Ah right, I guess we could :-)